asset-transfer-private-data/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,412 @@
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~');
+
+        // Create new assets on the ledger.
+        await createAsset(contractOrg1);
+
+        //Read  asset from the org1's private data collection.
+        await readAssetPrivateDetails(contractOrg1,org1PrivateCollectionName,assetID1);
+
+        //Attempt to transfer asset without prior aprroval from org2, transaction expected to fail.
+        await transferAssetWithoutAprroval(contractOrg1);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org2 Client ~~~~~~~~~~~~~~~~');
+
+        //Read the asset by ID.
+        await readAssetByID(contractOrg2);
+
+        //Make agreement to transfer the asset from org1 to org2.
+        await agreeToTransfer(contractOrg2);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~');
+
+        //Read transfer agreement.
+        await readTransferAgreement(contractOrg1);
+
+        // Tranfer asset to org2.
+        await transferAsset(contractOrg1);
+
+        //Again ReadAsset : results will show that the buyer identity now owns the asset.
+        await readAssetByID(contractOrg1);
+
+        //Confirm that transfer removed the private details from the Org1 collection.
+        await checkPrivateDataExist(contractOrg1);
+
+        // Delete AssetID2 as Org1.
+        await deleteAsset(contractOrg1);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org2 Client ~~~~~~~~~~~~~~~~');
+
+        //Org2 can ReadAssetPrivateDetails: Org2 is owner, and private details exist in new owner's Collection
+        await readAssetPrivateDetails(contractOrg2,org2PrivateCollectionName,assetID1);
+
+    } finally {
+        gatewayOrg1.close();
+        clientOrg1.close();
+
+        gatewayOrg2.close();
+        clientOrg2.close();
+    }
+}
+
+main().catch((error) =>
+    console.error('******** FAILED to run the application:', error)
+);
+
+async function newGrpcConnection(
+    tlsCertPath: string,
+    peerEndpoint: string,
+    peerContainer: string
+): Promise<grpc.Client> {
+    const tlsRootCert = await fs.readFile(tlsCertPath);
+    const tlsCredentials = grpc.credentials.createSsl(tlsRootCert);
+    return new grpc.Client(peerEndpoint, tlsCredentials, {
+        'grpc.ssl_target_name_override': peerContainer,
+    });
+}
+
+async function newIdentity(certPath: string, mspId: string): Promise<Identity> {
+    const credentials = await fs.readFile(certPath);
+    return { mspId, credentials };
+}
+
+async function newSigner(keyDirectoryPath: string): Promise<Signer> {
+    const files = await fs.readdir(keyDirectoryPath);
+    const keyPath = path.resolve(keyDirectoryPath, files[0]);
+    const privateKeyPem = await fs.readFile(keyPath);
+    const privateKey = crypto.createPrivateKey(privateKeyPem);
+    return signers.newPrivateKeySigner(privateKey);
+}
+
+function doFail(msgString: string) {
+    console.error(`${RED}\t${msgString}${RESET}`);
+    process.exit(1);
```

Comment:
`proxess.exit()` is very aggressive. Console output may not get written before the process terminates so you may not see the errors. It might be better to thrown an error to terminate execution more cleanly